client/api/locks.ts

```diff
@@ -0,0 +1,319 @@
+    IFeaturedPlanningLock, IAssignmentItem,
+} from '../interfaces';
+import {planningApi, superdeskApi} from '../superdeskApi';
+
+import {EVENTS, LOCKS, PLANNING, WORKSPACE, ASSIGNMENTS} from '../constants';
+
+import featuredPlanning from '../actions/planning/featuredPlanning';
+import {lockUtils, getErrorMessage, eventUtils, planningUtils, isExistingItem} from '../utils';
+import {currentWorkspace as getCurrentWorkspace} from '../selectors/general';
+import {getLockedItems} from '../selectors/locks';
+
+function loadLockedItems(types?: Array<'events_and_planning' | 'featured_planning' | 'assignments'>): Promise<void> {
+    const {gettext} = superdeskApi.localization;
+    const {notify} = superdeskApi.ui;
+    let url = 'planning_locks';
+
+    if ((types?.length ?? 0) > 0) {
+        url += `?repos=${types.join(',')}`;
+    }
+
+    return superdeskApi.dataApi.queryRawJson<ILockedItems & {featured?: ILock}>(url).then(
+        (locks) => {
+            const {dispatch} = planningApi.redux.store;
+
+            dispatch({
+                type: LOCKS.ACTIONS.RECEIVE,
+                payload: locks,
+            });
+
+            // If `featured_planning` lock was retrieved, then update it's lock now
+            if (types == null || types.includes('featured_planning')) {
+                if (locks.featured?.user != null) {
+                    dispatch(featuredPlanning.setLockUser(locks.featured.user, locks.featured.session));
+                } else {
+                    dispatch(featuredPlanning.setUnlocked());
+                }
+            }
+
+            // Make sure that all items that are locked are loaded into the store
+            return planningApi.combined.searchGetAll({
+                item_ids: lockUtils.getLockedItemIds(locks),
+                only_future: false,
+                include_killed: true,
+                spike_state: 'draft',
+                exclude_rescheduled_and_cancelled: false,
+                include_associated_planning: true,
+            }).then(
+                (items) => {
+                    dispatch({
+                        type: EVENTS.ACTIONS.ADD_EVENTS,
+                        payload: items.filter((item) => item.type === 'event'),
+                    });
+                    dispatch({
+                        type: PLANNING.ACTIONS.RECEIVE_PLANNINGS,
+                        payload: items.filter((item) => item.type === 'planning'),
+                    });
+                },
+                (error) => {
+                    notify.error(getErrorMessage(error, gettext('Failed to load locked items')));
+
+                    return Promise.reject(error);
+                }
+            );
+        },
+        (error) => {
+            notify.error(getErrorMessage(error, gettext('Failed to load item locks')));
+
+            return Promise.reject(error);
+        }
+    );
+}
+
+function addLockToStore(data: IWebsocketMessageData['ITEM_LOCKED']): void {
```

Comment:
not sure if we should mention the store in the method name, that made me think it's related to localStorage or something, maybe `setLockedState` would do